### DISCUSSION TITLE
Vue is listed as a programming language but It´s a javascript framework

### DISCUSSION THREAD

**Discussion by claudioantonio** [github-linguist/linguist#5257] [General]
## Describe the bug
Vue is listed as a programming language but It is a javascript framework as ReactJS. But ReactJS is not listed as a programming language. 🤔 
For example: https://api.github.com/repos/PanJiaChen/vue-element-admin/languages

### Expected behaviour
Vue should not be returned as a programming language.

### Related discussion
https://github.com/darensin01/goodfirstissues/issues/31

**Comment by Alhadis** [comment]
[What would you call this then?](https://github.com/PanJiaChen/vue-element-admin/blob/1bc2e5c1980b2e5131164849f66dd4f03216806d/src/layout/components/Sidebar/index.vue)

**Comment by claudioantonio** [comment]
Hi @Alhadis ! IMHO, there is 2 points to address in Your question:

1. In terms of programming language definition returning Javascript for the case You provided would be a good answer as It is predominant in this file. This same question arise with a [ReactJS code](https://github.com/claudioantonio/territoryFrontend/blob/master/src/page/Register/Register.tsx). But differently from Vue, ReactJS files don´t have to use a specific file extension (typically .js, .tsx or .jsx) as Vue files have. Even return HTML as the answer would be acceptable given that there is a DIV tag in this file.
2. How difficult is to make precise definition for the programming language used in web projects where style (css), layout structure (HTML and specialized tags) and scripting (js, ts, etc) can all be put together in the same file?

**Reply by lildude** [reply]
Both your points are assuming a level of complexity in the functionality that Linguist doesn't have and probably never will.

The limitation you're encountering is Linguist does not, and probably never will, perform multi-language analysis of the same file as it would be too resource intensive and fraught with inaccuracies for very little gain. Linguist is designed to give a "good enough" breakdown of the languages in a repo so it analyses a repository's language breakdown on a per-file basis without considering the rest of the files and their languages. It allows for overrides for those cases where it doesn't get it right or the user has differing opinions or preferences.

**Comment by lildude** [comment]
As this is a discussion, I'm moving this to "Discussions"
